filetablemodel.cpp

```diff
@@ -106,10 +107,23 @@ QVariant FileTableModel::data(const QModelIndex &index,
                               int role) const
 {
     // We validate arguments.
-    if (!index.isValid() || role != Qt::DisplayRole ||
-            index.column() >= columnCount() || index.row() >= rowCount())
+    if (!index.isValid() || index.column() >= columnCount() || index.row() >= rowCount())
+        return QVariant();
+    switch (role) {
+    case Qt::DisplayRole:
+        return m_data.at(index.column()).at(index.row());
+        break;
+    case Qt::CheckStateRole:
+        if ((index.column() == 0) && (index.row() < m_checkedStates.count())) {
+            return m_checkedStates[index.row()];
+        }
+        else {
```

Comment:
Ok, let's keep to the opening brace on the same line rule from now on. Regarding the unary operators, you don't have to enclose them in parentheses.